ecosystem/lava-sdk/src/stateTracker/stateQuery/state_chain_query.ts

```diff
@@ -120,14 +125,50 @@ export class StateChainQuery {
           continue;
         }
 
+        downtimeParams = pairingResponse.getDowntimeParams();
+
         const providers = pairing.getProvidersList();
         timeLeftToNextPairing = pairing.getTimeLeftToNextPairing();
-        const currentEpoch = pairingResponse.getPairing()?.getCurrentEpoch();
+        currentEpoch = pairingResponse.getPairing()?.getCurrentEpoch();
         if (!currentEpoch) {
           throw Logger.fatal(
             "Failed fetching current epoch from pairing request."
           );
         }
+
+        // check if epoch has not changed
+        if (
```

Comment:
this shouldn't be in the update pairing chains loop, and should move to an emergency tracker class that you can test separately and have the code more similar to the go implementation